engine/common/follower/engine.go

```diff
@@ -302,7 +302,7 @@ func (e *Engine) processBlockProposal(proposal *messages.BlockProposal) error {
 	// check whether the block is a valid extension of the chain.
 	// it only checks the block header, since checking block body is expensive.
 	// The full block check is done by the consensus participants.
-	err := e.state.Mutate().HeaderExtend(block)
+	err := e.state.Extend(block)
```

Comment:
because it got refactored to `Extend`. `Extend` implements or follower logic(previous `HeaderExtend`) and full consensus logic obviously depending on instantiated object.